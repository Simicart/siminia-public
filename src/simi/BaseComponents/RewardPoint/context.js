// import React, { createContext, useContext } from 'react';
// import { useRewardPoint } from './talons/useRewardPoint';

// const BssRewardPointContext = createContext();
// const { Provider } = BssRewardPointContext;

// const BssRewardPointProvider = props => {
//     const { children } = props;

//     const talonProps = useRewardPoint();
//     const { error } = talonProps;

//     if (error) {
//         if (process.env.NODE_ENV !== 'production') {
//             console.error(error);
//         }
//     }

//     const contextValue = {
//         ...talonProps
//     };

//     return <Provider value={contextValue}>{children}</Provider>;
// };

// export default BssRewardPointProvider;

// export const useAmMegaMenuContext = () => useContext(BssRewardPointContext);
