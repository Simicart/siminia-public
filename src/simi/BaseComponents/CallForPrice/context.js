import React, { createContext, useContext } from 'react';
import { useCallForPrice } from './talons/useCallForPrice';

const CallForPriceContext = createContext();
const { Provider } = CallForPriceContext;

const CallForPriceProvider = props => {
    const { children } = props;

    const talonProps = useCallForPrice();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default CallForPriceProvider;

export const useCallForPriceContext = () => useContext(CallForPriceContext);
