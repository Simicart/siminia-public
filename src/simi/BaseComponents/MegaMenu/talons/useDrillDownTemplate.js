import { useAmMegaMenuContext } from '../context';
import { useEffect, useMemo } from 'react';
import { buildTree, mapItems } from '../utils';
import { useLazyQuery } from '@apollo/client';
import DEFAULT_OPERATIONS from '@magento/peregrine/lib/talons/CategoryTree/categoryTree.gql';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';

export const useDrillDownTemplate = props => {
  const { categoryId, updateCategories } = props;
  const { allItems, config, isShowIcons } = useAmMegaMenuContext();

  const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
  const { getNavigationMenuQuery } = operations;

  const [runQuery, queryResult] = useLazyQuery(getNavigationMenuQuery, {
    fetchPolicy: 'no-cache',
    // nextFetchPolicy: 'cache-first'
  });
  const { data } = queryResult;

  // fetch categories
  useEffect(() => {
    if (categoryId != null) {
      runQuery({ variables: { id: categoryId } });
    }
  }, [categoryId, runQuery]);

  // update redux with fetched categories
  useEffect(() => {
    if (data && data.category) {
      updateCategories(data.category);
    }
  }, [data, updateCategories]);

  const { root_category_id: rootCategoryId } = config;
  const isTopLevel = categoryId === rootCategoryId;

  const categories = useMemo(() => {
    if (!Array.isArray(allItems)) {
      return null;
    }

    const tree = buildTree(mapItems(allItems), true);

    if (isTopLevel) {
      return [...tree.values()].filter(({ parent_id }) => !parent_id);
    }

    const category = tree.get(categoryId);
    // const { subCategories, name } = category;
    const subCategories = category && category.subCategories ? category.subCategories : ''
    const name = category && category.name ? category.name : ''

    return [
      { ...category, name: `All ${name}`, subCategories: null },
      ...subCategories
    ];
  }, [allItems, categoryId, isTopLevel]);

  return {
    categories,
    isShowIcons,
    isTopLevel
  };
};
