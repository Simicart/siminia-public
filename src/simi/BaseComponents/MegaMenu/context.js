import React, { createContext, useContext } from 'react';
import { useMegaMenu } from './talons/useMegaMenu';

const AmMegaMenuContext = createContext();
const { Provider } = AmMegaMenuContext;

const AmMegaMenuProvider = props => {
    const { children } = props;

    const talonProps = useMegaMenu();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default AmMegaMenuProvider;

export const useAmMegaMenuContext = () => useContext(AmMegaMenuContext);
