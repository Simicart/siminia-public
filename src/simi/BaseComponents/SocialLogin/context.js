import React, { createContext, useContext } from 'react';
import { useSocialLogin } from './talons/useSocialLogin';
import { useSignInEvent } from './talons/useSignInEvent';

const AmSocialLoginContext = createContext();
const { Provider } = AmSocialLoginContext;

const AmAdvancedReviewsProvider = props => {
  const { children } = props;

  const talonProps = useSocialLogin();
  const { error, needRedirect, redirectUrl, handleErrors } = talonProps;
  const { isBusy } = useSignInEvent({
    needRedirect,
    redirectUrl,
    handleErrors
  });

  if (error) {
    if (process.env.NODE_ENV !== 'production') {
      console.error(error);
    }

    console.log('Check if Amasty modules has been installed!');
  }

  const contextValue = {
    ...talonProps,
    isBusy
  };

  return <Provider value={contextValue}>{children}</Provider>;
};

export default AmAdvancedReviewsProvider;

export const useSocialLoginContext = () => useContext(AmSocialLoginContext);
