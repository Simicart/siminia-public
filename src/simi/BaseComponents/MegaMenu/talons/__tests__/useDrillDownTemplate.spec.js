import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import { useLazyQuery } from '@apollo/client';
import { useDrillDownTemplate } from '../useDrillDownTemplate';
import { useAmMegaMenuContext } from '../../context';

jest.mock('../../context', () => ({
  useAmMegaMenuContext: jest.fn()
}));

jest.mock('@apollo/client', () => {
  const apolloClient = jest.requireActual('@apollo/client');
  const useLazyQuery = jest.fn();

  return { ...apolloClient, useLazyQuery };
});

const mockItems = [
  {
    name: 'Bottoms',
    id: 'category-node-11',
    url: 'venia-bottoms.html',
    parent_id: null,
    __typename: 'MenuCategoryItem'
  },
  {
    name: 'Pants & Shorts',
    id: 'category-node-12',
    url: 'venia-bottoms/venia-pants.html',
    parent_id: 'category-node-11'
  }
];

const mockConfig = {
  root_category_id: 2
};

const defaultProps = {
  categoryId: 2,
  updateCategories: jest.fn()
};

const mockRunQuery = jest.fn();

const mockCategoryData = {
  called: true,
  error: null,
  loading: false,
  data: {
    category: {
      id: 11,
      children: [
        {
          children_count: 1,
          id: 12
        }
      ]
    }
  }
};

const Component = props => {
  const talonProps = useDrillDownTemplate(props);

  return <i {...talonProps} />;
};

test('return correct tree first level', () => {
  useAmMegaMenuContext.mockReturnValue({
    allItems: mockItems,
    config: mockConfig
  });
  useLazyQuery.mockReturnValue([mockRunQuery, mockCategoryData]);

  const rendered = createTestInstance(<Component {...defaultProps} />);
  const talonProps = rendered.root.findByType('i').props;
  const { categories } = talonProps;
  expect(mockRunQuery).toHaveBeenCalledTimes(1);
  expect(categories).toHaveLength(1);
  expect(categories).toMatchSnapshot();
});

test('return correct tree second level', () => {
  useAmMegaMenuContext.mockReturnValue({
    allItems: mockItems,
    config: mockConfig
  });
  useLazyQuery.mockReturnValue([mockRunQuery, mockCategoryData]);

  const rendered = createTestInstance(
    <Component {...defaultProps} categoryId={11} />
  );
  const talonProps = rendered.root.findByType('i').props;
  const { categories } = talonProps;
  expect(categories).toHaveLength(2);
  expect(categories).toMatchSnapshot();
});
