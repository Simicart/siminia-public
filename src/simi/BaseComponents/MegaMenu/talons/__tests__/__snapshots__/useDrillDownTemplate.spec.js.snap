// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return correct tree first level 1`] = `
Array [
  Object {
    "__typename": "MenuCategoryItem",
    "id": 11,
    "name": "Bottoms",
    "parentId": null,
    "parent_id": null,
    "subCategories": Array [
      Object {
        "id": 12,
        "name": "Pants & Shorts",
        "parentId": 11,
        "parent_id": 11,
        "url": "venia-bottoms/venia-pants.html",
      },
    ],
    "url": "venia-bottoms.html",
  },
]
`;

exports[`return correct tree second level 1`] = `
Array [
  Object {
    "__typename": "MenuCategoryItem",
    "id": 11,
    "name": "All Bottoms",
    "parentId": null,
    "parent_id": null,
    "subCategories": null,
    "url": "venia-bottoms.html",
  },
  Object {
    "id": 12,
    "name": "Pants & Shorts",
    "parentId": 11,
    "parent_id": 11,
    "url": "venia-bottoms/venia-pants.html",
  },
]
`;
